src/neptune_scale/net/api_client.py

```diff
@@ -131,9 +138,43 @@ def create_auth_api_client(
         follow_redirects=True,
         verify_ssl=verify_ssl,
         timeout=Timeout(timeout=HTTP_CLIENT_NETWORKING_TIMEOUT),
+        headers={"User-Agent": _generate_user_agent()},
     )
 
 
+def _generate_user_agent() -> str:
+    import platform
+    from importlib.metadata import version
+
+    package_name = "neptune-scale"
+    package_version = "unknown"
+    additional_metadata = {
+        "neptune-api": "unknown",
+        "python": "unknown",
+        "os": "unknown",
+    }
+
+    try:
+        package_version = version(package_name)
+    except Exception:
+        pass
+    try:
+        additional_metadata["neptune-api"] = version("neptune-api")
+    except Exception:
+        pass
+    try:
+        additional_metadata["python"] = platform.python_version()
+    except Exception:
+        pass
+    try:
+        additional_metadata["os"] = platform.platform()
```

Comment:
IMO if we hope for the backend to parse this, we need to make sure it follows some protocol: either it cannot contain `();` or it cannot contain spaces. 

TBH i'd just go with `platform.system()` and still take care to replace spaces with underscores.